### PR TITLE
Added Docker resources for ION

### PR DESCRIPTION
**Pull request by caleteeter**
Added a core docker automation via docker compose and a script.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  ```
+
+- Docker compose (latest)
+  ```
+  sudo curl -L --max-time 10 --retry 3 --retry-delay 3 --retry-max-time 60 "https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+
+  sudo chmod +x /usr/local/bin/docker-compose
+  ```
+
+- NodeJS - version 10.x   `NOTE: This has a requirement for version 10 of nodejs only`
```
Comment (thehenrytsai) [inline docker/README.md:43]: What is the cause for requiring specifically Node v10.x?

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    volumes:
+      - ${DATA_VOL}:/data
+  bitcoin-ion:
+    container_name: bitcoin-ion
+    image: node:10.20.1-alpine3.11
+    command: "npm run bitcoin"
+    working_dir: /ion
+    links:
+      - "bitcoin-core"
+      - "mongo"
```
Comment (thehenrytsai) [inline docker/docker-compose.yml:22]: Sanity: Is there a need to specify `links`? From [links documentation](https://docs.docker.com/compose/compose-file/#links) it appears that usage is not necessary. Are you using it to enforce dependency start up sequence? Also it says `link` is legacy and should be avoided if possible.

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    volumes:
+      - ./ion:/ion
+  ipfs:
+    container_name: ipfs
+    image: node:10.20.1-alpine3.11
+    command: "npm run ipfs"
+    working_dir: /ion
```
Comment (thehenrytsai) [inline docker/docker-compose.yml:29]: This is an interesting approach, different to what I was thinking, but certainly easier to compose... I was thinking of having images for ION services. Do you see this as THE way to compose and run? Or this is more of a short term solution?

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+- bitcoin-core - This container will run the bitcoin-core client.  `NOTE: This is currenlty using testnet as the source chain`
+
+- bitcoin-ion - This is the bitcoin sidetree node for ION.
+
+- ipfs - This is the local IPFS node used by the sidetree.
+
+- mongo - This is the local MongoDB used by the sidetree.
+
+- core-ion - This is the core sidetree node.
```
Comment (thehenrytsai) [inline docker/README.md:13]: Nit: our convention have been:

ion-bitcoin
ion-ipfs
ion-core

File: docker/README.md (hunk 1 above)
Comment (caleteeter) [reply docker/README.md:43]: No hard requirement from me, but it was stated in the install guide [here](https://github.com/decentralized-identity/ion/blob/master/install-guide.md) to use NodeJS v10.  I have not tested with 12, but would be good to support this.

File: docker/docker-compose.yml (hunk 2 above)
Comment (caleteeter) [reply docker/docker-compose.yml:22]: Link is used here to allow name resolution for dependent containers.  I am not 100% sure that depends_on will give you this without setting up a specific network.  Basically this allows the internal network and name resolution without the need to create a network.

File: docker/docker-compose.yml (hunk 3 above)
Comment (caleteeter) [reply docker/docker-compose.yml:29]: Short term solution, for publishing ION containers, it makes sense to wrap a nodejs container with each one for ION services.  This also means that we would either need to redirect the configuration for these containers to a volume or distributed configuration specific to each service.  The latter seems to be the clean way to promote reuse.

From a scaling perspective, having these all be separate independent containers makes sense.  For instance I can use a shared IPFS instance and not use this container.

File: docker/README.md (hunk 4 above)
Comment (caleteeter) [reply docker/README.md:13]: Naming can be changed of course, sorry if these names were not the write scheme.

File: docker/README.md (hunk 1 above)
Comment (csuwildcat) [reply docker/README.md:43]: @thehenrytsai sounds like we need to update the guide to list v12? Are there any other things to change in the guide along with this?

Comment (thehenrytsai) [reply docker/README.md:43]: @csuwildcat, yes, we should change to v12. I don't know if there are other changes required as I have not gone through the installation guide for a while, but we do try to update them when we make changes that we know impact the installation process..

Comment (caleteeter) [reply docker/README.md:43]: @thehenrytsai  any update, ok to use v12 nodejs?

**Pull request merged**